lib/src/metta/runner/stdlib.rs

```diff
@@ -622,16 +630,22 @@ impl Display for SuperposeOp {
 
 impl Grounded for SuperposeOp {
     fn type_(&self) -> Atom {
-        Atom::expr([ARROW_SYMBOL, ATOM_TYPE_UNDEFINED, ATOM_TYPE_UNDEFINED])
+        Atom::expr([ARROW_SYMBOL, ATOM_TYPE_EXPRESSION, ATOM_TYPE_UNDEFINED])
     }
 
     fn execute(&self, args: &mut Vec<Atom>) -> Result<Vec<Atom>, ExecError> {
         let arg_error = || ExecError::from("superpose expects single expression as an argument");
-        // `superpose` receives one atom (expression) in order to make composition
-        // `(superpose (collapse ...))` possible
-        let expr = atom_as_expr(args.get(0).ok_or_else(arg_error)?).ok_or(arg_error())?;
+        let atom = args.get(0).ok_or_else(arg_error)?;
+        let expr = atom_as_expr(&atom).ok_or(arg_error())?;
 
-        Ok(expr.children().clone())
+        let mut superposed = Vec::new();
+        for atom in expr.children() {
+            match interpret_no_error(self.space.clone(), atom) {
+                Ok(results) => { superposed.extend(results); },
+                Err(message) => { return Err(format!("Error: {}", message).into()) },
```

Comment:
Should `Err` be returned here, or should it be added to `superposed`? What are implications of these two approaches?

Reply:
In current version this branch will not be executed because `interpret` never returns `Err` in fact. May be it is more logical to replace it with `panic!`. I returned `Err` here as I would expect in future `interpret` may use `Err` to return some errors which are cannot be handled on a MeTTa code level. But may be it doesn't make sense.

Reply:
I would say it is a separate task to understand whether we need `Err` in `interpret`.

Reply:
To summarize: I would leave it as is or add a `panic!` here, but anyway we need to revisit this code while revisiting `interpreter` API.

Reply:
OK, got it. I thought `Err` can appear because of badly typed expressions or other errors in MeTTa code, but it seems that MeTTa `Err` expressions will already be produced instead of Rust `Err`. So, let's leave it as is.